evaluator/builtin_other.go

```diff
@@ -25,6 +26,28 @@ import (
 	"github.com/pingcap/tidb/util/types"
 )
 
+// See http://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_sleep
+func builtinSleep(args []types.Datum, _ context.Context) (d types.Datum, err error) {
+	if args[0].IsNull() {
```

Comment:
MySQL will treat null or negative numbers as zero.
mysql> select sleep(null);
+-------------+
| sleep(null) |
+-------------+
|           0 |
+-------------+
1 row in set (0.00 sec)

mysql> select sleep(-1);
+-----------+
| sleep(-1) |
+-----------+
|         0 |
+-----------+
1 row in set (0.00 sec)